src/build/mdn-comments.ts

```diff
@@ -0,0 +1,85 @@
+import fs from "fs/promises";
+import * as path from "path";
+
+const basePath = new URL(
+  "../../inputfiles/mdn/files/en-us/web/api/",
+  import.meta.url,
+).pathname;
+
+function extractSummary(markdown: string): string {
+  // Remove frontmatter (--- at the beginning)
+  markdown = markdown.replace(/^---[\s\S]+?---\n/, "");
+
+  // Normalize line breaks by collapsing consecutive newlines into a single space
+  const normalizedText = markdown
+    .split("\n")
+    .map((line) => line.trim())
+    .filter(
+      (line) =>
+        line &&
+        !line.startsWith("#") &&
+        !line.startsWith(">") &&
+        !line.startsWith("{{"),
+    )
+    .join(" ");
+
+  // Extract the first sentence (ending in . ! or ?)
+  const sentenceMatch = normalizedText.match(/(.*?[.!?])(?=\s|$)/);
+  if (sentenceMatch) {
+    return sentenceMatch[0]; // Return the first full sentence
+  }
+
+  return normalizedText.split(" ")[0] || ""; // Fallback: first word if no sentence found
+}
+
+async function getFolders(dirPath: string): Promise<string[]> {
+  try {
+    const entries = await fs.readdir(dirPath, { withFileTypes: true });
+    return entries
+      .filter((entry) => entry.isDirectory())
+      .map((entry) => path.join(dirPath, entry.name));
```

Comment:
Here for example, this could be `new URL(entry.name, dirPath)` and return `Promise<URL[]>`.